users/app.py

```diff
@@ -54,22 +54,50 @@ def add_update_user():
     return jsonify(user_object), (201 if added_new_user else 200)
 
 
+def upsert_user_in_db(user_object, users_collection):
+    """Updates or inserts the user object into users_collection.
+
+    Sets/overwrites default values as needed.
+
+    Args:
+        user_object (dict): must contain a user_id and name, and may contain
+            other attributes. Default values will be set as needed.
+        users_collection (pymongo.collection): the MongoDB collection to use.
+
+    Returns:
+        ObjectID: The ID of the upserted object from the db. This can be used
+            to find the object with collection.find_one(object_id).
+
+    Raises:
+        AttributeError: if user_object is malformatted.
+    """
+    if user_object.keys() < {"user_id", "name"}:
+        raise AttributeError("missing user_id or name")
+    # insert is_organizer field (default False)
+    user_object["is_organizer"] = False
```

Comment:
You may want to make a copy of the object here so you aren't modifying the passed-in input.

Reply:
Done https://github.com/knative-portability/large-events/pull/8/commits/80bbf7263589bdc41a637f599b31f281c314b6f5